### PR TITLE
feat: deprecate upcoming ECMAScript 2024 and unused functions

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
* **#146** <a href="https://app.graphite.dev/github/pr/skyleague/axioms/146?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/skyleague/axioms/146?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/array/_deprecated/find-last/find-last.ts [hunk 1]
```diff
@@ -35,6 +35,7 @@ import { reverse } from '../reverse/index.js'
  * @typeParam T - The element type.
  *
  * @group Array
+ * @deprecated Use `xs.findLast(by)` instead.
```
Comment (MickVanDuijn) [inline src/array/_deprecated/find-last/find-last.ts:38]: this doesn't exist, does it? :grin:

File: src/array/_deprecated/head/head.ts [hunk 2]
```diff
@@ -31,6 +31,7 @@ import type { Traversable } from '../../type/traversable/index.js'
  * @typeParam T - The element type.
  *
  * @group Array
+ * @deprecated Use `xs[0]` instead.
```
Comment (MickVanDuijn) [inline src/array/_deprecated/head/head.ts:34]: what about iterators?

File: src/array/_deprecated/reverse/reverse.ts [hunk 3]
```diff
@@ -30,6 +30,7 @@ import type { Traversable } from '../../type/index.js'
  * @typeParam T - The element type.
  *
  * @group Array
+ * @deprecated Use `xs.reverse()` instead.
```
Comment (MickVanDuijn) [inline src/array/_deprecated/reverse/reverse.ts:33]: what about iterators?

File: src/generator/_deprecated/range/range.ts [hunk 4]
```diff
@@ -28,6 +28,7 @@ import { isDefined } from '../../guard/index.js'
  * @returns The range generator.
  *
  * @group Generators
+ * @deprecated While this function is really nice, using native ECMAScipt is prefered.
```
Comment (MickVanDuijn) [inline src/generator/_deprecated/range/range.ts:31]: do we have a "nice" alternative in ECMA?

File: src/iterator/_deprecated/group-by/group-by.ts [hunk 5]
```diff
@@ -29,6 +29,7 @@ import { foldl } from '../fold/index.js'
  * @typeParam K - The key type.
  *
  * @group Iterators
+ * @deprecated Use `xs.groupBy(group)` instead
```
Comment (MickVanDuijn) [inline src/iterator/_deprecated/group-by/group-by.ts:32]: this does not exist, does it? haven't found something like this in arrays/iterators

**Review by MickVanDuijn — APPROVED** [review]

File: src/array/_deprecated/find-last/find-last.ts (hunk 1 above)
Comment (pevisscher) [reply src/array/_deprecated/find-last/find-last.ts:38]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLast

File: src/iterator/_deprecated/group-by/group-by.ts (hunk 5 above)
Comment (pevisscher) [reply src/iterator/_deprecated/group-by/group-by.ts:32]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy

File: src/array/_deprecated/reverse/reverse.ts (hunk 3 above)
Comment (pevisscher) [reply src/array/_deprecated/reverse/reverse.ts:33]: `xs.toArray().reverse()`, that's what would have happened anyway

**Comment by pevisscher** [comment]
### Merge activity

* **Dec 10, 1:52 PM GMT+1**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/skyleague/axioms/146).

**Pull request merged**